spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -1042,10 +1042,10 @@ object CometSparkSessionExtensions extends Logging {
    *   The node with information (if any) attached
    */
   def withInfo[T <: TreeNode[_]](node: T, info: String, exprs: T*): T = {
-    val exprInfo = exprs
-      .flatMap { e => Seq(e.getTagValue(CometExplainInfo.EXTENSION_INFO)) }
-      .flatten
-      .mkString("\n")
+    val exprInfo =
+      (Seq(node.getTagValue(CometExplainInfo.EXTENSION_INFO)) ++ exprs
+        .flatMap(e => Seq(e.getTagValue(CometExplainInfo.EXTENSION_INFO)))).flatten
+        .mkString("\n")
```

Comment:
@parthchandra I found an issue where if I call `withInfo` twice on the same node, it would lose the info from the first call so I modified this to preserve any existing tag on the node.

Does this seem reasonable?

Reply:
Hmm I guess this will end up duplicating info. I will rethink this and add some tests.

Reply:
I updated this to avoid duplicates

Reply:
I no longer need this for this PR so will file a separate issue to discuss this